python/lsst/ts/adamSensors/adamSensorsCSC.py

```diff
@@ -44,9 +40,23 @@ async def begin_start(self, data):
         self.cmd_start.ack_in_progress(data, timeout=self.start_timeout)
         await super().begin_start(data)
         self.adam = AdamModel(self.log, simulation_mode=self.simulation_mode)
-        self.adam.connect(self.params.ip, self.params.port)
+        try:
+            await self.adam.connect(self.config.adam_ip, self.config.adam_port)
+        except ConnectionException:
+            self.fault(code=2, report="Unable to connect to modbus device at " + str(self.config.adam_ip))
```

Comment:
Uhm... I don't think is is what you should do. fault is reserved for when something goes unexpectedly wrong. In this case you probably want reject the command and stay in the current state.

The easiest way to do this is not do anything. If the call:

```
await self.adam.connect(self.config.adam_ip, self.config.adam_port)
```

raises an exception, the command will be rejected. You may want to capture and thread the exception to tell the user what happened, e.g.;

```
        try:
            await self.adam.connect(self.config.adam_ip, self.config.adam_port)
        except ConnectionException:
            raise RuntimeError(f"Unable to connect to modbus device at {self.config.adam_ip}:{self.config.adam_port}.")
```